contrib/globalsign/mgo/mgo_test.go

```diff
@@ -0,0 +1,355 @@
+package mgo
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func testMongoCollectionCommand(assert *assert.Assertions, command func(*Collection)) []mocktracer.Span {
```

Comment:
This is one idea that could compact all tests into table-tests:
```go
func TestCollection(t *testing.T) {
    for name, tt := range map[string]struct{
        op   func(*Collection)
        test func(*assert.Assertions, []mocktracer.Span)
    } {
        "insert-update": {
            op: func(c *Collection) error {
                if err := c.Insert(e0); err != nil {
                    return err
                }
                if err := c.Update(e0, e1); err != nil { // let's vary on entities (not always the same)
                    return err
                }
            },
            assertion: func(assert *assert.Assertions, spans []mocktracer.Span, msg string) {
                assert.Equal(3, len(spans), msg)
                // etc. (use msg)
            },
        },

        // continue with other tests

    } {
        t.Run(name, func(t *testing.T) {
            mt := mocktracer.Start()
            defer mt.Stop()

            collection, stopFn := connect() // connects to server, returns collection and stop function
            defer stopFn() // closes db connection and stops mock tracer
            if err := tt.command(collection); err != nil {
                t.Fatal(err)
            }
            tt.assertion(assert.New(t), mt.FinishedSpans())
        })
    }
}
```
We should test a bigger variety too (not always the same entity) and check that all tags are also set correctly (resource, service name, etc).